mep3_localization/mep3_localization/robot_detection.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+import rclpy
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+import cv2
+import numpy as np
+from cv2 import aruco
+from cv_bridge import CvBridge
+from tf2_ros import TransformBroadcaster
+from scipy.spatial.transform import Rotation as R
+import math
+
+import tf_transformations
+from geometry_msgs.msg import TransformStamped
+"""
+This class receives video from the central RasPi Cam,
+and sends TF of Aruco tags.
+"""
+
+
+class DetectedRobots(Node):
+
+    def __init__(self):
+        super().__init__('image_subscriber')
+        self.subscription = self.create_subscription(
+            Image, '/cam/cam_central/RasPi0', self.listener_callback, 10)
+        self.subscription
+        self._tf_publisher = TransformBroadcaster(self)
+
+        # cv_file = cv2.FileStorage('calibration_camera.yaml', cv2.FILE_STORAGE)
+
+        self.br = CvBridge()
+        self.dict = aruco.Dictionary_get(aruco.DICT_4X4_100)
+        self.params = aruco.DetectorParameters_create()
+        # self.cameraMatrix = cv_file.getNode('K').mat()
+        # self.distCoeffs = cv_file.getNode('D').mat()
+        self.cameraMatrix = np.array([[570.34, 0, 1920 / 2],
+                                      [0, 570.34, 1080 / 2], [0, 0, 1]])
+        self.distCoeffs = np.zeros((5, 1))
+        # Side length of the ArUco marker in meters
+        self.aruco_robot_length = 0.07
+        self.aruco_map_length = 0.1
+        # cv_file.release()
+
+    def listener_callback(self, data):
+        # self.get_logger().info('Receiving video frame')
+        current_frame = self.br.imgmsg_to_cv2(data, "bgr8")
+        self.draw_aruco_pose(current_frame)
+
+    def make_transforms(self, tvec, rvec, id):
+        static_transformStamped = TransformStamped()
+        static_transformStamped.header.stamp = self.get_clock().now().to_msg()
+        static_transformStamped.header.frame_id = 'world'
+        static_transformStamped.child_frame_id = 'robot' + str(id)
```

Comment:
```suggestion
        static_transform_stamped.header.frame_id = 'map'
        static_transform_stamped.child_frame_id = f'marker_{id}'
```
- underscore notacija (`static_transform_stamped`)
- koristimo `map` kao globalni frejm u ostatku sistema pa da budem konzistentni
- mozda mozemo objaviti sve markere?

Reply:
Možemo objaviti sve markere, ali će nam verovatno trebati neki algoritam za klasifikaciju, odnosno praćenje pozicije markera. Ne znam koliko je to jednostavno.